synapse/replication/tcp/handler.py

```diff
@@ -124,12 +142,13 @@ def start_replication(self, hs):
         # We protect catching up with a linearizer in case the replication
         # connection reconnects under us.
         with await self._position_linearizer.queue(cmd.stream_name):
-            # We're about to go and catch up with the stream, so mark as connecting
-            # to stop RDATA being handled at the same time by removing stream from
-            # list of connected streams. We also clear any batched up RDATA from
-            # before we got the POSITION.
+            # We're about to go and catch up with the stream, so remove from set
+            # of connected streams.
             self._streams_connected.discard(cmd.stream_name)
```

Comment:
is this actually doing anything useful? once we've caught up for the first time, `_streams_connected[cmd.stream_name]` is always set outside the linearizer. Alternatively expressed: the linearizer ensures that no RDATA will be processed while we catch up, so there is no need for us to clear `_streams_connected`. I'd just get rid of these three lines.

Reply:
Thinking about this a bit, the only difference is if an exception is raised. I don't think we want to handle RDATA for that stream if we fail to handle the position, but dropping everything doesn't sound like the right thing either

Reply:
mmmm this sounds like a thing we need to improve, but ok let's punt it for now.